aws_s3_policies/aws_s3_bucket_secure_access.py

```diff
@@ -11,9 +19,24 @@ def policy(resource):
     iam_policy = Policy(json.loads(resource["Policy"]))
 
     for statement in iam_policy.statements:
-        if statement.effect != "Allow":
-            continue
-        if not deep_get(statement.statement, "Condition", "Bool", "aws:SecureTransport"):
-            return False
+        if (
+            statement.effect == "Deny"
+            and deep_get(statement.statement, "Condition", "Bool", "aws:SecureTransport") == "false"
+        ):
+            return True
+
+        if (
```

Comment:
The only issue I see is we must check all effect==allow statements to make sure they contain secureTransport==true before returning true in that case. If an explicit deny statement doesn't exist and not all Allow statements have secureTransport==true then the policy should technically fail. Please tell me if I am wrong but that was how I understood it (I am also not extremely well versed in S3 access policies)

Reply:
Good call, I'll update accordingly